node/src/components/consensus/protocols/highway.rs

```diff
@@ -240,6 +249,21 @@ impl<I: NodeIdT, C: Context> HighwayProtocol<I, C> {
         now: Timestamp,
     ) -> Vec<CpResult<I, C>> {
         let start_time = Timestamp::now();
+        // Check whether we should change the round exponent.
+        // It's important to do it before the vertex is added to the state - this way if the last
+        // round has finished, we now have all the vertices from that round in the state, and no
+        // newer ones.
+        if vv.inner().value().is_some() {
```

Comment:
Can you extract `vv.inner().value().is_some()` to a method that is self-descriptive? Probably `is_proposal(vv)`.